asmcli/components/ca/managed-cas.sh

```diff
@@ -75,6 +75,54 @@ x_enable_workload_certificate_on_membership() {
      -d "${ENABLEFEATURE}" "https://${GKEHUB_API}/v1alpha/projects/${FLEET_ID}/locations/global/features/workloadcertificate?update_mask=membership_specs"
 }
 
+x_wait_for_enabling_workload_certificates() {
+  local GKEHUB_API; GKEHUB_API="$1"
+  local FLEET_ID; FLEET_ID="$2"
+
+  info "Waiting for the workload certificates enablement. This may take up to 20 minutes ..."
+  # Enabling workload certificate feature on a cluster will result in
+  # the cluster restarting and the cluster becoming unreachable for minutes.
+  # The cluster restart may happen at a random time after the enablement.
+  # Therefore, wait 15 minutes before checking the connection to k8s is recovered.
+  sleep 900
+  # Check the connection to k8s is recovered.
+  local REACHABLE; REACHABLE=false
+  for i in {1..3}; do
+    local OUT; OUT=$(kubectl version -o json 2>/dev/null | jq .serverVersion.major -r 2>/dev/null)
+    if [[ $OUT -eq 1 ]]; then
+      info "The k8s cluster is reachable."
+      REACHABLE=true
+      break
+    else
+      info "The k8s cluster not reachable, try again ..."
+      sleep 60
+    fi
+  done
+  if ! $REACHABLE; then
+    fatal "The k8s cluster is not reachable, exit."
+  fi
+
+  # Check the status of workload certificate feature
+  local ENABLED; ENABLED=false
+  for i in {1..2}; do
+    local OUT; OUT="$(curl -H "X-Goog-User-Project: ${FLEET_ID}" \
+      -H "Authorization: Bearer $(gcloud auth print-access-token)" \
+      "https://${GKEHUB_API}/v1alpha/projects/${FLEET_ID}/locations/global/features/workloadcertificate" 2>/dev/null | \
+      grep -i -c "certificateManagement.*ENABLED")"
+    if [[ $OUT -gt 0 ]]; then
```

Comment:
is OUT supposed to be a string? could we just not do a -z here?

Reply:
IMO, this check should come before the previous one - the CLH propagation happens reasonably quickly. Its only the restart of the cluster that can take some time...